oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/PriorityCache.java

```diff
@@ -265,22 +309,27 @@ public synchronized boolean put(@NotNull K key, @NotNull V value, int generation
      */
     @SuppressWarnings("unchecked")
     @Nullable
-    public synchronized V get(@NotNull K key, int generation) {
+    public V get(@NotNull K key, int generation) {
         int hashCode = key.hashCode();
         for (int k = 0; k <= rehash; k++) {
             int i = project(hashCode, k);
+            Segment segment = getSegment(i);
+            segment.lock();
             Entry<?, ?> entry = entries[i];
             if (generation == entry.generation && key.equals(entry.key)) {
                 if (entry.cost < Byte.MAX_VALUE) {
-                    costs[entry.cost - Byte.MIN_VALUE]--;
+                    costs[entry.cost - Byte.MIN_VALUE].decrementAndGet();
                     entry.cost++;
-                    costs[entry.cost - Byte.MIN_VALUE]++;
+                    costs[entry.cost - Byte.MIN_VALUE].incrementAndGet();
                 }
-                hitCount++;
-                return (V) entry.value;
+                hitCount.incrementAndGet();
+                V value = (V) entry.value;
+                segment.unlock();
+                return value;
             }
+            segment.unlock();
```

Comment:
In case of exception, `unlock()` will not be called. It's safer to call it in a `finally` clause. The `try` block should start just after `lock()` is called. The call to `unlock()` at line 327 can be removed.